src/bh_db_worker.erl

```diff
@@ -38,46 +48,75 @@ squery(Pool, Sql) ->
 equery(Pool, Stmt, Params) ->
     case do_checkout(Pool, ?POOL_CHECKOUT_RETRIES) of
         {ok, Reference, Conn} ->
-            Res = epgsql:equery(Conn, Stmt, Params),
-            dispcount:checkin(Pool, Reference, Conn),
-            Res;
+            case epgsql:parse(Conn, Stmt) of
+                {ok, #statement{types = Types} = Statement} ->
+                    TypedParameters = lists:zip(Types, Params),
+                    Ref = epgsqla:equery(Conn, Statement, TypedParameters),
+                    dispcount:checkin(Pool, Reference, Conn),
+                    receive
+                        {Conn, Ref, Res} ->
+                            Res
+                    after
+                        500 ->
```

Comment:
Is there some magic in the 500 value? Do we need to create a constant here?

Reply:
I just made up a number